clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java

```diff
@@ -63,6 +63,12 @@ public boolean isNullable() {
         return false;
     }
 
+    /**
+     * Documentation of the Type.
+     * @return details about valid values, representation
+     */
+    public abstract String documentation();
```

Comment:
Just a suggestion, but this could also be put in a sub-class (say `DocumentedType`), which `Schema` need not extend from.

Reply:
I like this idea.